### PR TITLE
fix: --state-values-set unable to set booleans

### PR DESCRIPTION
**Pull request by tunahansezen**
This pr fixes auto-wrapping of booleans and integers into quotes when using --state-values-set by:

- Adding: --state-values-set-string flag for intentional string set of boolean or integer
- Changing: --state-values-set flag not wrapping now
- Removing -

Resolves https://github.com/roboll/helmfile/issues/1347

### CODE REVIEW

**Comment by yxxhero** [comment]
@tunahansezen please fix DCO issue.

File: pkg/maputil/maputil.go [hunk 1]
```diff
@@ -187,5 +188,55 @@ func Set(m map[string]any, key []string, value string) {
 		m, key = getCursor(key[0]).getMap(m), key[1:]
 	}
 
-	getCursor(key[0]).set(m, value)
+	getCursor(key[0]).set(m, typedVal(value, stringBool))
+}
+
+func typedVal(val string, st bool) any {
```
Comment (yxxhero) [inline pkg/maputil/maputil.go:194]: please add some comments and unit tests for this function.

File: pkg/maputil/maputil.go [hunk 2]
```diff
@@ -187,5 +188,55 @@ func Set(m map[string]any, key []string, value string) {
...
+	if len(val) != 0 && val[0] != '0' {
+		if iv, err := strconv.ParseInt(val, 10, 64); err == nil {
+			return iv
+		}
+	}
+
+	return val
+}
+
+func MergeMaps(a, b map[string]interface{}) map[string]interface{} {
```
Comment (yxxhero) [inline pkg/maputil/maputil.go:225]: please add some comments and unit tests for this function. @tunahansezen

File: pkg/maputil/maputil.go (hunk 1 above)
Comment (tunahansezen) [reply pkg/maputil/maputil.go:194]: I've added some tests for all types and added some comments for it.

File: pkg/maputil/maputil.go (hunk 2 above)
Comment (tunahansezen) [reply pkg/maputil/maputil.go:227]: I've added a test and some comments to key points. In "helm" package, handled set-values with this method. I belive example in the test is sufficiently descriptive.

**Pull request merged**